src/main/java/ui/ScreenBuilder.java

```diff
@@ -0,0 +1,358 @@
+package ui;
+
+import check_profile_validation.CheckProfileIGateway;
+import check_profile_validation.CheckProfileInputBoundary;
+import check_profile_validation.CheckProfileInteractor;
+import controller.CheckProfileController;
+import data_access.CheckProfileDataAccess;
+import presenter.CheckProfilePresenter;
+import presenter.IViewModel;
+import presenter.UseCase;
+import view_model.IView;
+import view_model.Table;
+import view_model.ViewModel;
+
+import javax.swing.*;
+import javax.swing.table.DefaultTableModel;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.UUID;
+
+public class ScreenBuilder implements IView {
+    private final IViewModel dataModel;
+    private final Integration view;
+    private final CheckProfileController buttonController;
+
+    public ScreenBuilder(IViewModel dataModel) {
+        this.dataModel = dataModel;
+        view = new Integration(dataModel.getFrameName());
+        this.dataModel.addObserver(view);
+        CheckProfileIGateway gateway = new CheckProfileDataAccess();
+        CheckProfilePresenter presenter =new CheckProfilePresenter();
+        CheckProfileInputBoundary interactor = new CheckProfileInteractor(gateway, presenter);
+        buttonController = new CheckProfileController(interactor);
+    }
+
+    public ScreenBuilder(ViewModel dataModel, CheckProfileIGateway gateway) {
+
+        this.dataModel = dataModel;
+        // Add view to the dataModel observable.
+        view = new Integration(dataModel.getFrameName());
+        this.dataModel.addObserver(view);
+        CheckProfilePresenter presenter =new CheckProfilePresenter();
```

Comment:
Shouldn't CheckProfilePresenter take in the dataModel?

Reply:
presenter form the dataModel, which is the ViewModel in MVVP, and that dataModel will control the appearance of UI